src/rust-crypto/DehydratedDeviceManager.ts

```diff
@@ -166,19 +175,25 @@ export class DehydratedDeviceManager {
      *
      * @returns the key, if available, or `null` if no key is available
      */
-    private async getKey(create: boolean): Promise<Uint8Array | null> {
-        if (this.key === undefined) {
-            const keyB64 = await this.secretStorage.get(SECRET_STORAGE_NAME);
-            if (keyB64 === undefined) {
-                if (!create) {
-                    return null;
-                }
-                await this.resetKey();
-            } else {
-                this.key = decodeBase64(keyB64);
+    private async getKey(create: boolean): Promise<RustSdkCryptoJs.DehydratedDeviceKey | null> {
+        const cachedKey = await this.olmMachine.dehydratedDevices().getDehydratedDeviceKey();
+        if (cachedKey) return cachedKey;
+        const keyB64 = await this.secretStorage.get(SECRET_STORAGE_NAME);
+        if (keyB64 === undefined) {
+            if (!create) {
+                return null;
+            }
+            return await this.resetKey();
+        } else {
+            const bytes = decodeBase64(keyB64);
+            try {
+                const key = RustSdkCryptoJs.DehydratedDeviceKey.createKeyFromArray(bytes);
+                await this.cacheKey(key);
+                return key;
+            } finally {
+                bytes.fill(0);
             }
         }
```

Comment:
If you're going to do early returns, might as well take benefit from them

```suggestion

        // We successfully found the key in secret storage: decode it, and cache it in
        // the rust SDK's crypto store.
        const bytes = decodeBase64(keyB64);
        try {
            const key = RustSdkCryptoJs.DehydratedDeviceKey.createKeyFromArray(bytes);
            await this.cacheKey(key);
            return key;
        } finally {
            bytes.fill(0);
        }
```

Reply:
.oO (if we're going to `.fill(0)` on `bytes`, shouldn't we also do so for `keyB64` ?)

Reply:
> .oO (if we're going to `.fill(0)` on `bytes`, shouldn't we also do so for `keyB64` ?)

Ideally we should, but keyB64 is a string and I don't think we can modify a string?

Reply:
true